contracts/RenPool.sol

```diff
@@ -70,16 +77,44 @@ contract RenPool {
         uint senderBalance = balances[sender];
 
         require(senderBalance > 0 && senderBalance >= _amount, "Insufficient funds");
-        require(isLocked == false, "Pool is locked");
+        if(!isLocked){
+            totalPooled -= _amount;
+            balances[sender] -= _amount;
+            renToken.transfer(sender, _amount);
+            
+            emit RenWithdrawal(sender, _amount);
+
+        }
+        else{
+            // Pool is locked, withdraw will be put in the queue
+            withdrawRequests.push(WithdrawRequest(msg.sender, _amount));           
+        }
 
```

Comment:
I still think we should have a different function for the requestWithdraw functionality. Then the consumer of the contract (the frontend in our case) will decide which function to call. Also, we should add a new event for the requestWithdraw method.

Reply:
Okay yes that makes sence.
And regarding the events I was thinking making a new branch just for all the events because there are a bunch that we need to create in my opinion